android/modules/locale/src/java/ti/modules/titanium/locale/LocaleModule.java

```diff
@@ -82,6 +83,39 @@ public String formatTelephoneNumber(String telephoneNumber)
 		return PhoneNumberUtils.formatNumber(telephoneNumber);
 	}
 
+	@Kroll.method
+	public double parseDecimal(String text, @Kroll.argument(optional = true) String localeString)
+	{
+		double result = Double.NaN;
+		try {
+			// Create a number format parser using given locale if provided or current locale.
+			Locale locale = TiPlatformHelper.getInstance().getLocale(localeString);
+			NumberFormat numberFormat;
+			if (locale != null) {
+				numberFormat = NumberFormat.getInstance(locale);
+			} else {
+				numberFormat = NumberFormat.getInstance();
+			}
+
+			// Enable thousands separator parsing support. (ex: "1,234,567")
+			numberFormat.setGroupingUsed(true);
+
+			// Remove leading spaces and plus sign. Number format will fail to parse if there.
+			text = text.trim();
+			if ((text != null) && text.startsWith("+")) {
```

Comment:
Shouldn't need the null check since you're already trimming without checking above. I assume our bridge layer enforces non-null if the argument is not marked optional? If not, the null check needs to move to the very top.